src/test/java/org/mockito/internal/creation/instance/ConstructorInstantiatorTest.java

```diff
@@ -40,13 +47,33 @@ public void creates_instances_of_inner_classes() {
     }
 
     @Test
+    public void creates_instances_with_arguments() {
+        assertEquals(new ConstructorInstantiator("someString").newInstance(SomeClass2.class).getClass(), SomeClass2.class);
+    }
+
+    @Test
+    public void creates_instances_with_null_arguments() {
+        assertEquals(new ConstructorInstantiator(new Object[]{null}).newInstance(SomeClass2.class).getClass(), SomeClass2.class);
+    }
+
+    @Test
+    public void creates_instances_with_primitive_arguments() {
+        assertEquals(new ConstructorInstantiator(123).newInstance(SomeClass3.class).getClass(), SomeClass3.class);
+    }
+
+    @Test(expected = InstantiationException.class)
```

Comment:
Based on the feedback above, the error messages refer to the passed argument types, not their values. `null`s, unfortunately, don't have types, so there's no way I can conclude it.  I'd be happy to improve this, but I'm not quite sure what you're suggesting here. Could you elaborate please?